FinniversKit/Sources/Components/PromotionView/PromotionView.swift

```diff
@@ -107,33 +130,72 @@ public class PromotionView: UIView {
         titleLabel.text = viewModel.title
         primaryButton.configure(withTitle: viewModel.primaryButtonTitle)
         secondaryButton.configure(withTitle: viewModel.secondaryButtonTitle)
+        imageRatio = viewModel.image.size.width / viewModel.image.size.height
 
         if let text = viewModel.text {
             textLabel.text = text
         } else {
             textLabel.isHidden = true
         }
 
+        if let backgroundImage = viewModel.backgroundImage {
+            backgroundImageContainer.addSubview(backgroundImageView)
+            backgroundImageView.image = backgroundImage
+            backgroundImageView.fillInSuperview()
+        }
+
         imageContainer.addSubview(imageView)
         imageView.image = viewModel.image
 
         switch viewModel.imageAlignment {
         case .fullWidth:
             imageView.fillInSuperview(insets: UIEdgeInsets(top: .spacingM, leading: .spacingS, bottom: -.spacingM, trailing: -.spacingS))
         case .trailing:
-            let imageRatio = viewModel.image.size.width / viewModel.image.size.height
+            guard let imageRatio = imageRatio else { return }
```

Comment:
It's a bit "dangerous" this LOC depends on us remembering to set the `imageRatio` above ☝️
It used to be required and now it's optional. 
What if we make image ratio an extension of the view model, like

```
private extension PromotionViewModel {
     var imageRatio: CGFloat {
          image.size.width / image.size.height
```
The constraints above can also use the view model property, like `viewModel?.imageRatio ?? 1`, as long as we save the view model when configuring it 😄  
```
    func configure(with viewModel: PromotionViewModel) {
         self.viewModel = viewModel
```

Reply:
Cool 👍

Reply:
Actually the viewModel can be saved on init of the view, so it doesn't even have to be optional. Much better.

Reply:
perfect ❤️